### PR TITLE
Camel openstack

### PR DESCRIPTION
**Pull request by bouskaJ**
New component to work with openstack. 
This pull request is related to [CAMEL-9748](https://issues.apache.org/jira/browse/CAMEL-9748)

### CODE REVIEW

**Comment by oscerd** [comment]
Will review and merge soon.

@davsclaus about docs regenerated, do you know if we did some upgrade on the tooling. There are a lot of Docs stuff removed in this build..

**Comment by bouskaJ** [comment]
It seems that I did something wrong - I rebuild it again today and the stuff from doc is not removed now (somewhere it is reworded but not removed). I will fix it tomorrow.

**Comment by bouskaJ** [comment]
I pushed the results of the new build but there are still some lines removed. Should I fix it manually?

**Comment by tadayosi** [comment]
I'm just curious why the API Component Framework hasn't been used with it.
http://planet.jboss.org/post/apache_camel_api_component_framework_or_how_i_learned_to_stop_worrying_and_love_writing_feature_rich_camel_components

**Comment by oscerd** [comment]
@davsclaus do you think the docs generated is ok?

**Comment by davsclaus** [comment]
Not sure have not looked at that, but could be his build was not clean etc. We can do a rebuild after merging a PR or add those back manually.

**Comment by oscerd** [comment]
Ok, I'll start the review.

File: components/camel-openstack/src/test/java/org/apache/camel/component/openstack/neutron/NeutronProducerTestSupport.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package org.apache.camel.component.openstack.neutron;
```
Comment (oscerd) [inline components/camel-openstack/src/test/java/org/apache/camel/component/openstack/neutron/NeutronProducerTestSupport.java]: Please add Apache license header

File: components/camel-openstack/src/test/java/org/apache/camel/component/openstack/neutron/PortProducerTest.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
+package org.apache.camel.component.openstack.neutron;
```
Comment (oscerd) [inline components/camel-openstack/src/test/java/org/apache/camel/component/openstack/neutron/PortProducerTest.java]: Please add Apache license header

File: components/camel-openstack/src/test/java/org/apache/camel/component/openstack/neutron/RouterProducerTest.java [hunk 3]
```diff
@@ -0,0 +1,205 @@
+package org.apache.camel.component.openstack.neutron;
```
Comment (oscerd) [inline components/camel-openstack/src/test/java/org/apache/camel/component/openstack/neutron/RouterProducerTest.java]: Please add Apache license header

File: components/camel-openstack/src/test/java/org/apache/camel/component/openstack/neutron/SubnetProducerTest.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
+package org.apache.camel.component.openstack.neutron;
```
Comment (oscerd) [inline components/camel-openstack/src/test/java/org/apache/camel/component/openstack/neutron/SubnetProducerTest.java]: Please add Apache license header

File: components/camel-openstack/src/test/java/org/apache/camel/component/openstack/nova/FlavorProducerTest.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
+package org.apache.camel.component.openstack.nova;
```
Comment (oscerd) [inline components/camel-openstack/src/test/java/org/apache/camel/component/openstack/nova/FlavorProducerTest.java]: Please add Apache license header

File: components/camel-openstack/src/test/java/org/apache/camel/component/openstack/nova/KeypairProducerTest.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
+package org.apache.camel.component.openstack.nova;
```
Comment (oscerd) [inline components/camel-openstack/src/test/java/org/apache/camel/component/openstack/nova/KeypairProducerTest.java]: Please add Apache license header

File: components/camel-openstack/src/test/java/org/apache/camel/component/openstack/nova/NovaProducerTestSupport.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
+package org.apache.camel.component.openstack.nova;
```
Comment (oscerd) [inline components/camel-openstack/src/test/java/org/apache/camel/component/openstack/nova/NovaProducerTestSupport.java]: Please add Apache license header

File: components/camel-openstack/src/test/java/org/apache/camel/component/openstack/nova/ServerProducerTest.java [hunk 8]
```diff
@@ -0,0 +1,139 @@
+package org.apache.camel.component.openstack.nova;
```
Comment (oscerd) [inline components/camel-openstack/src/test/java/org/apache/camel/component/openstack/nova/ServerProducerTest.java]: Please add Apache license header

File: components/camel-openstack/src/test/java/org/apache/camel/component/openstack/swift/ContainerProducerTest.java [hunk 9]
```diff
@@ -0,0 +1,177 @@
+package org.apache.camel.component.openstack.swift;
```
Comment (oscerd) [inline components/camel-openstack/src/test/java/org/apache/camel/component/openstack/swift/ContainerProducerTest.java]: Please add Apache license header

File: components/camel-openstack/src/test/java/org/apache/camel/component/openstack/swift/ObjectProducerTest.java [hunk 10]
```diff
@@ -0,0 +1,175 @@
+package org.apache.camel.component.openstack.swift;
```
Comment (oscerd) [inline components/camel-openstack/src/test/java/org/apache/camel/component/openstack/swift/ObjectProducerTest.java]: Please add Apache license header

File: components/camel-openstack/src/test/java/org/apache/camel/component/openstack/swift/SwiftProducerTestSupport.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
+package org.apache.camel.component.openstack.swift;
```
Comment (oscerd) [inline components/camel-openstack/src/test/java/org/apache/camel/component/openstack/swift/SwiftProducerTestSupport.java]: Please add Apache license header

File: components/camel-openstack/pom.xml [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+		<groupId>org.apache.camel</groupId>
+		<artifactId>components</artifactId>
+		<version>2.19.0-SNAPSHOT</version>
+	</parent>
+
+	<artifactId>camel-openstack</artifactId>
+	<packaging>jar</packaging>
+	<name>Camel :: OpenStack</name>
+
+	<properties>
```
Comment (oscerd) [inline components/camel-openstack/pom.xml]: Please move the version properties in parent/pom.xml

**Comment by oscerd** [comment]
Added review @bouskaJ 

Thanks! :-)

**Comment by oscerd** [comment]
@bouskaJ can you add references to the JIRA in commit messages? Thanks

**Comment by davsclaus** [comment]
@bouskaJ are you able to work on these reviews and update the PR?

**Comment by bouskaJ** [comment]
@davsclaus Sure, I am working on it right now.

**Comment by bouskaJ** [comment]
@oscerd It should be fixed.

**Comment by oscerd** [comment]
@bouskaJ Thanks the PR has been merged. Can you close this? :-) Many thanks!

**Comment by bouskaJ** [comment]
Sure, I will close this.

**Pull request closed without merging**